### ISSUE TITLE
WAFileMetadataLibrary subclasses assumes ASCII files when importing CSS/JS

### ISSUE THREAD

**Issue by astares** [SeasideSt/Seaside#1273]
Sanjay Minni reports on Pharo Discord #seaside channel that he downloaded the 5.1 bundle of Bootstrap (including CSS and JS files) and while importing on a subclass of WAFileMetadataLibrary he received an encoding error.

![image](https://user-images.githubusercontent.com/5980033/128389542-3ecb500e-120b-4fdd-afe5-53d5e5fee143.png)

To reproduce extract https://github.com/twbs/bootstrap/releases/download/v5.1.0/bootstrap-5.1.0-dist.zip
into a folder. Create a new subclass of **WAFileMetadataLibrary **:

```Smalltalk
WAFileMetadataLibrary subclass: #MyFileLibrary
	instanceVariableNames: ''
	classVariableNames: ''
	package: 'Sample-Package'
```	
and import using 
```Smalltalk
MyFileLibrary recursivelyAddAllFilesIn: 'C:\temp\bootstrap-5.1.0-dist'
```

This will crash at position 9035 of the file stream with a character value 226

![image](https://user-images.githubusercontent.com/5980033/128390192-a20d8e08-9acf-4e8b-86df-168208afbd98.png)

**Comment by astares** [comment]
I guess the problem is that the platform files assume that the encoding of the files is ASCII while todays CSS and JS files are typically writen as UTF-8 encoded files.

![image](https://user-images.githubusercontent.com/5980033/128390567-21906022-deed-462a-a786-f5f7b5eb8c67.png)

If the method **GRPharoPlatform>>#readFileStreamOn:do:binary:** is temporarily patched as 

```Smalltalk
readFileStreamOn: aString do: aBlock binary: aBoolean

	^ aBoolean
		ifTrue: [ aString asFileReference binaryReadStreamDo: aBlock ]
		ifFalse: [ aString asFileReference readStreamEncoded: 'utf-8' do: aBlock ]
```

it works without a problem.

**Comment by jbrichau** [comment]
Thanks for reporting. 
I guess it's safe to classify ccs files as utf8. This is not by patching that method but by asserting that utf8 is the encoding for the css mimetype. I'm currently starting a holiday so I don't know when I will be able to get to this, but that's the way to go.

**Comment by jbrichau** [comment]
On second thought, because utf-8 is actually backwards compatible with ascii, it does make sense to just replace the 'ascii' encoding with 'utf-8' encoding in that method. 

Transferred the issue to Grease: https://github.com/SeasideSt/Grease/issues/122

**Comment by jbrichau** [comment]
So, I discovered that although the code in Grease has always stipulated 'ascii' encoding, the underlying implementation in Pharo prior to Grease 1.6.0 did use utf-8 encoding for text files. So, this is a regression since Grease 1.6.0 where we moved to the new filestream api in Pharo.

See https://github.com/SeasideSt/Grease/issues/122

A new Grease bugfix release is almost ready to be published.

**Comment by jbrichau** [comment]
Fixed in Grease 1.7.3 
See https://github.com/SeasideSt/Grease/releases/tag/v1.7.3
